### PR TITLE
Fix 3613 :function atan with two parameters

### PR DESCRIPTION
**Pull request by YANGGMM**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3613

## What this PR does / why we need it:
function atan with two parameters

### CODE REVIEW

**Comment by ouyuanning** [comment]
why not just rewrite  atan(y, x) to atan(y/x) ？  in plan or in function register.

that will make logic more simple

File: pkg/sql/plan/function/builtin/unary/math.go [hunk 1]
```diff
@@ -62,7 +62,37 @@ func Acos(vs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
...
+	if len(vs) == 1 {
+		return math1(vs, proc, momath.Atan)
+	}
+	//Otherwise 
+	//First if one of the parameters is a null value
+	typ := vs[0].Typ.Oid.ToType() 
+	if vs[0].IsScalarNull() || vs[1].IsScalarNull(){
+		return proc.AllocScalarNullVector(typ), nil
+	}
+
```
Comment (ouyuanning) [inline pkg/sql/plan/function/builtin/unary/math.go:75]: I think some code like that will work?
newVs, err := Arith[T, T](vs, proc, vs[0].GetType(), div.NumericDivFloat[T])
if err != nil {
    return err
}
return math1(vs, proc, momat.Atan)

File: pkg/sql/plan/function/builtin/unary/math.go [hunk 2]
```diff
@@ -62,7 +62,37 @@ func Acos(vs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
...
+	//Otherwise 
+	//First if one of the parameters is a null value
+	typ := vs[0].Typ.Oid.ToType() 
+	if vs[0].IsScalarNull() || vs[1].IsScalarNull(){
+		return proc.AllocScalarNullVector(typ), nil
+	}
+	
+	//Second if the vs is scalar
+	if vs[0].IsScalar() {
+		resultVector := proc.AllocScalarVector(types.Type{Oid: types.T_float64, Size: 8})
```
Comment (ouyuanning) [inline pkg/sql/plan/function/builtin/unary/math.go:78]: use types.T_float64.ToType()

File: pkg/sql/plan/function/builtin/unary/math.go [hunk 3]
```diff
@@ -62,7 +62,37 @@ func Acos(vs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
...
+		resultValues := make([]float64, 1)
```
Comment (ouyuanning) [inline pkg/sql/plan/function/builtin/unary/math.go:79]: if vs[1] can be vector. why resultValues's length is 1?

File: pkg/sql/plan/function/builtin/unary/math.go [hunk 4]
```diff
@@ -62,7 +62,37 @@ func Acos(vs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
...
+		vecLen := int64(vector.Length(vs[0]))
+		resultVector, err := proc.AllocVectorOfRows(types.T_float64.ToType(), vecLen, vs[0].Nsp)
+		if err != nil {
+			return nil, err
+		}
+		if err := momath.AtanWithTwoArg(vs[0], vs[1], resultVector); err != nil {
+			return nil, err
+		}
+		return resultVector, nil
+	}
```
Comment (ouyuanning) [inline pkg/sql/plan/function/builtin/unary/math.go:95]: if you want to use your AtanWithTwoArg.
you can code like:
if vs[0].IsScalar() && vs[1].IsScalar() {
      // make scalar response and return
}

if vs[0].IsScalar() {
     // use vs[1] to set length and get resultVector
}
if vs[1].IsScalar() {
     // use vs[0] to set length and get resultVector
}
if err := momath.AtanWithTwoArg(vs[0], vs[1], resultVector); err != nil {
     return nil, err
}

File: pkg/sql/plan/function/builtin/unary/math.go (hunk 2 above)
Comment (YANGGMM) [reply pkg/sql/plan/function/builtin/unary/math.go:78]: ok

File: pkg/sql/plan/function/builtin/unary/math.go (hunk 4 above)
Comment (YANGGMM) [reply pkg/sql/plan/function/builtin/unary/math.go:95]: ok

File: pkg/sql/plan/function/builtin/unary/math.go (hunk 1 above)
Comment (YANGGMM) [reply pkg/sql/plan/function/builtin/unary/math.go:75]: i have solved the problem in this way

**Review by aunjgr — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Pull request merged**